airflow/providers/common/io/xcom/backend.py

```diff
@@ -152,7 +152,7 @@ def deserialize_value(
 
         try:
             p = ObjectStoragePath(path) / XComObjectStoreBackend._get_key(data)
-            return json.load(p.open("rb", compression="infer"), cls=XComDecoder)
+            return json.load(p.fs.open(p.path, mode="rb", compression="infer"), cls=XComDecoder)
```

Comment:
I think we need to keep the `Path.open(...)` API.

Reply:
I agree. Any reason to use `fs.open` directly here @ap-- ?

Reply:
This is subtle.
`UPath.open()` in `v0.2.0+` right now has the identical method signature to `pathlib.Path.open()`. Arguably this could be considered a regression.

`UPath.fs` gives direct access to the fsspec filesystem, without this limitation.

To address it here, I added a `v0.1.4` style `ObjectStoragePath.open` method, that passes kwargs through to AbstractFileSystem.open. This restores the tests, and keeps behavior for ObjectStoragePath consistent